### ISSUE TITLE
Updating .NET to 4.6

### ISSUE THREAD

**Issue by xavier-morera** [SolrNet/SolrNet#283]
I am currently working on updating the .NET version of all projects to 4.6. Can anyone interested take a look at the latest commit and let me know if they have any suggestions on the issue I am facing?

**Comment by xavier-morera** [comment]
This would be the error
```

Creating .\merged
lib\ilmerge.exe /out:"merged\SolrNet.dll" /target:"fake.ilmergehelper+targetkind
" /internalize:"ilmerge.exclude" /xmldocs /lib:"SolrNet\bin\debug" /lib:"SolrNet
.DSL\bin\debug" /lib:"HttpWebAdapters\bin\debug" /lib:"Castle.Facilities.SolrNet
Integration\bin\debug" /lib:"Ninject.Integration.SolrNet\bin\debug" /lib:"Struct
ureMap.SolrNetIntegration\bin\debug" /lib:"AutofacContrib.SolrNet\bin\debug" /li
b:"Unity.SolrNetIntegration\bin\debug" /lib:"NHibernate.SolrNet\bin\debug" SolrN
et\bin\debug\SolrNet.dll SolrNet.DSL.dll HttpWebAdapters.dll Castle.Facilities.S
olrNetIntegration.dll Ninject.Integration.SolrNet.dll StructureMap.SolrNetIntegr
ation.dll AutofacContrib.SolrNet.dll Unity.SolrNetIntegration.dll NHibernate.Sol
rNet.dll
An exception occurred during merging:
Unresolved assembly reference not allowed: System.Core.
   at System.Compiler.Ir2md.GetAssemblyRefIndex(AssemblyNode assembly)
   at System.Compiler.Ir2md.GetTypeRefIndex(TypeNode type)
   at System.Compiler.Ir2md.VisitReferencedType(TypeNode type)
   at System.Compiler.Ir2md.GetMemberRefIndex(Member m)
   at System.Compiler.Ir2md.GetMethodToken(Method m)
   at System.Compiler.Ir2md.VisitMethodCall(MethodCall call)
   at System.Compiler.Ir2md.VisitMethodCall(MethodCall call)
   at System.Compiler.Ir2md.VisitAssignmentStatement(AssignmentStatement assignm
ent)
   at System.Compiler.Ir2md.VisitBlock(Block block)
   at System.Compiler.Ir2md.VisitBlock(Block block)
   at System.Compiler.Ir2md.VisitMethodBody(Method method)
   at System.Compiler.Ir2md.VisitMethod(Method method)
   at System.Compiler.Ir2md.VisitClass(Class Class)
   at System.Compiler.Ir2md.VisitModule(Module module)
   at System.Compiler.Ir2md.SetupMetadataWriter(String debugSymbolsLocation)
   at System.Compiler.Ir2md.WritePE(Module module, String debugSymbolsLocation,
BinaryWriter writer)
   at System.Compiler.Writer.WritePE(String location, Boolean writeDebugSymbols,
 Module module, Boolean delaySign, String keyFileName, String keyName)
   at System.Compiler.Writer.WritePE(CompilerParameters compilerParameters, Modu
le module)
   at ILMerging.ILMerge.Merge()
   at ILMerging.ILMerge.Main(String[] args)
Running build failed.
```

**Comment by gjunge** [comment]
Just downloaded this version, did a quick test. Ran into one issue before being able to run the build script:

I had to change the `lib/FAKE.exe.config` to include : `[ProgramFilesX86]\Microsoft SDKs\F#\4.1\Framework\v4.0\` in the `FSIPath` value. 
On Windows 10/VS 2017 that is the new path for F#

After that I could run it , and I could replicate your issue.

Finding this article [ILMerge - Unresolved assembly reference not allowed: System.Core](http://geekswithblogs.net/michelotti/archive/2010/06/02/ilmerge---unresolved-assembly-reference-not-allowed-system.core.aspx?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed%3A+geekswithblogs+(Geekswithblogs.net)) I created lib/ILMerge.exe.config with the following content:

``` xml
<configuration>
    <startup useLegacyV2RuntimeActivationPolicy="true">
        <requiredRuntime safemode="true" imageVersion="v4.0.30319" version="v4.0.30319"/>
    </startup>
</configuration>

This caused the ILMerge not to fail on that same System.Core error, but now I get a timeout.

I'll poke around a bit more.

**Comment by xavier-morera** [comment]
@gjunge thanks to you, we have this

![image](https://user-images.githubusercontent.com/7174278/29796034-3c3499ec-8c0c-11e7-8c13-e978870f69b1.png)

**Comment by gjunge** [comment]
xavier-morera, that's great, happy to help!

**Comment by xavier-morera** [comment]
Released in 0.7.1 and updated in Nuget
